ee/clickhouse/sql/dead_letter_queue.py

```diff
@@ -0,0 +1,108 @@
+from ee.clickhouse.sql.clickhouse import KAFKA_COLUMNS, REPLACING_MERGE_TREE, kafka_engine, table_engine, ttl_period
+from ee.kafka_client.topics import KAFKA_DEAD_LETTER_QUEUE
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE
+
+# We pipe our Kafka dead letter queue into CH for easier analysis and longer retention
+# This allows us to explore errors and replay events with ease
+
+DEAD_LETTER_QUEUE_TABLE = "events_dead_letter_queue"
+
+DEAD_LETTER_QUEUE_TABLE_BASE_SQL = """
+CREATE TABLE {table_name} ON CLUSTER {cluster}
+(
+    id UUID,
+    event_uuid UUID,
+    event VARCHAR,
+    properties VARCHAR,
+    distinct_id VARCHAR,
+    team_id Int64,
+    elements_chain VARCHAR,
+    created_at DateTime64(6, 'UTC'),
+    ip VARCHAR,
+    site_url VARCHAR,
+    now DateTime64(6, 'UTC'),
+    raw_payload VARCHAR,
+    error_timestamp DateTime64(6, 'UTC'),
+    error_location VARCHAR,
+    error VARCHAR
+    {extra_fields}
+) ENGINE = {engine}
+"""
+
+DEAD_LETTER_QUEUE_TABLE_SQL = (
+    DEAD_LETTER_QUEUE_TABLE_BASE_SQL
+    + """ORDER BY (id, event_uuid, distinct_id, team_id)
+{ttl_period}
+SETTINGS index_granularity=512
+"""
+).format(
+    table_name=DEAD_LETTER_QUEUE_TABLE,
+    cluster=CLICKHOUSE_CLUSTER,
+    extra_fields=KAFKA_COLUMNS,
+    engine=table_engine(DEAD_LETTER_QUEUE_TABLE, "_timestamp", REPLACING_MERGE_TREE),
+    ttl_period=ttl_period("_timestamp", 4),  # 4 weeks
+)
+
+KAFKA_DEAD_LETTER_QUEUE_TABLE_SQL = DEAD_LETTER_QUEUE_TABLE_BASE_SQL.format(
+    table_name="kafka_" + DEAD_LETTER_QUEUE_TABLE,
```

Comment:
Do we have some naming convention regarding table names?

Reply:
For the base tables not really. For kafka tables we always prepend `kafka_` and for materialized views we append `_mv`